System/Process.hs

```diff
@@ -639,6 +650,10 @@ getProcessExitCode ph@(ProcessHandle _ delegating_ctlc) = do
     Just e | was_open && delegating_ctlc -> endDelegateControlC e
     _                                    -> return ()
   return m_e
+    where getHandle :: ProcessHandle__ -> PHANDLE
+          getHandle (OpenHandle        h) = h
+          getHandle (ClosedHandle      _) = error "getHandle: handle closed."
```

Comment:
Not a hard requirement, but I'd prefer to avoid the partiality here. I was thinking something like:

```haskell
case getHandle p_ of
  Left e -> return (p_, (Just e, False))
  Right h -> do ...
where
  getHandle (OpenHandle h) = Right h
  getHandle (ClosedHandle e) = Left e
  getHandle (OpenExtHandle h _ _) = Right h
```